scripts/portconfig

```diff
@@ -41,59 +46,67 @@ class portconfig(object):
 
-    def set_speed(self, port, speed):
+    def set_speed(self, table_name, port, speed):
         if self.verbose:
             print("Setting speed %s on port %s" % (speed, port))
-        self.db.mod_entry(PORT_TABLE_NAME, port, {PORT_SPEED_CONFIG_FIELD_NAME: speed})
+        self.db.mod_entry(table_name, port, {PORT_SPEED_CONFIG_FIELD_NAME: speed})
 
-    def set_fec(self, port, fec):
+    def set_fec(self, table_name, port, fec):
         if self.verbose:
             print("Setting fec %s on port %s" % (fec, port))
-        self.db.mod_entry(PORT_TABLE_NAME, port, {PORT_FEC_CONFIG_FIELD_NAME: fec})
+        self.db.mod_entry(table_name, port, {PORT_FEC_CONFIG_FIELD_NAME: fec})
 
-    def set_mtu(self, port, mtu):
+    def set_mtu(self, table_name, port, mtu):
         if self.verbose:
             print("Setting mtu %s on port %s" % (mtu, port))
-        self.db.mod_entry(PORT_TABLE_NAME, port, {PORT_MTU_CONFIG_FIELD_NAME: mtu})
+        self.db.mod_entry(table_name, port, {PORT_MTU_CONFIG_FIELD_NAME: mtu})
+
 
 def main():
     parser = argparse.ArgumentParser(description='Set SONiC port parameters',
-                         formatter_class=argparse.RawTextHelpFormatter)
+                                     formatter_class=argparse.RawTextHelpFormatter)
     parser.add_argument('-p', '--port', type=str, help='port name (e.g. Ethernet0)', required=True, default=None)
-    parser.add_argument('-l', '--list', action='store_true', help='list port parametars', default=False)
+    parser.add_argument('-l', '--list', action='store_true', help='list port parameters', default=False)
     parser.add_argument('-s', '--speed', type=int, help='port speed value in Mbit', default=None)
     parser.add_argument('-f', '--fec', type=str, help='port fec mode value in (none, rs, fc)', default=None)
     parser.add_argument('-m', '--mtu', type=int, help='port mtu value in bytes', default=None)
     parser.add_argument('-v', '--version', action='version', version='%(prog)s 1.0')
     parser.add_argument('-vv', '--verbose', action='store_true', help='Verbose output', default=False)
-    parser.add_argument('-n', '--namespace', metavar='namespace details', type = str, required = False,
-                        help = 'The asic namespace whose DB instance we need to connect', default=None)
+    parser.add_argument('-n', '--namespace', metavar='namespace details', type=str, required=False,
+                        help='The asic namespace whose DB instance we need to connect', default=None)
     args = parser.parse_args()
 
     if args.namespace is not None:
         swsssdk.SonicDBConfig.load_sonic_global_db_config(namespace=args.namespace)
 
     try:
-        port = portconfig(args.verbose, args.port, args.namespace)
+        if args.port.startswith(PORT_PREFIX_ETHERNET):
+            port_table_name = PORT_TABLE_NAME
+        elif args.port.startswith(PORT_PREFIX_PORTCHANNEL):
+            port_table_name = PORTCHANNEL_TABLE_NAME
```

Comment:
This logic can also be replaced with https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-py-common/sonic_py_common/interface.py#L69

Reply:
Done.